libarchive/archive_write_disk_posix.c

```diff
@@ -1981,6 +1981,10 @@ restore_entry(struct archive_write_disk *a)
 	if ((en == EISDIR || en == EEXIST)
 	    && (a->flags & ARCHIVE_EXTRACT_NO_OVERWRITE)) {
 		/* If we're not overwriting, we're done. */
+		if (S_ISDIR(a->mode)) {
```

Comment:
Why is this only for directories?  It seems like `NO_OVERWRITE` should avoid changing permissions, owner, flags, etc, on regular files as well.

Reply:
I only included directories in this change because it's the more straight-forward case to deal with. For example, xattrs could get cleared on regular files when the contents are modified but the flags aren't restored, resulting in a change anyway. Solving that problem felt out of scope for this change.